manufacturing_order_type/models/mrp_production.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Ecosoft Co., Ltd (http://ecosoft.co.th/).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def _default_order_type(self):
+        return self.env["manufacturing.order.type"].search(
+            ["|", ("company_id", "=", False), ("company_id", "=", self.env.company.id)],
+            limit=1,
+        )
+
+    order_type = fields.Many2one(
+        comodel_name="manufacturing.order.type",
+        string="Type",
+        ondelete="restrict",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+        default=_default_order_type,
+    )
+
+    @api.onchange("product_id")
+    def onchange_product_id(self):
```

Comment:
```suggestion
    def _onchange_product_id(self):
```

Reply:
Why do I have to change from `onchange_product_id` to `_onchange_product_id` when the core use `onchange_product_id`?

Reply:
My point was to make this method private.

Reply:
I agree with @sswapnesh it is what we normally use.
@niaisoh can you fix it in your commit?